src/dataDisplay/tag/tag.jsx

```diff
@@ -0,0 +1,339 @@
+    },
+});
+
+class Label extends React.Component {
+    constructor() {
+        super();
+
+        this.onClearClick = this.onClearClick.bind(this);
+        this.onClearKeyDown = this.onClearKeyDown.bind(this);
+        this.onClick = this.onClick.bind(this);
+        this.onKeyDown = this.onKeyDown.bind(this);
+    }
+
+    onClearClick() {
+        const {
+            onClearClick,
+        } = this.props;
+
+        if (isFunction(onClearClick)) {
+            onClearClick();
+        }
+    }
+
+    onClearKeyDown() {
+        const {
+            onClearKeyDown,
+        } = this.props;
+
+        if (isFunction(onClearKeyDown)) {
+            onClearKeyDown();
+        }
+    }
+
+    onClick() {
+        const {
+            onClick,
+        } = this.props;
+
+        if (isFunction(onClick)) {
+            onClick();
+        }
+    }
+
+    onKeyDown() {
+        const {
+            onKeyDown,
+        } = this.props;
+
+        if (isFunction(onKeyDown)) {
+            onKeyDown();
+        }
+    }
+
+    render() {
+        const {
+            children,
+            classes,
+            className,
+            color,
+            fluid,
+            inverse,
+            onClick,
+            onClearClick,
+            style,
+            tabIndex,
+        } = this.props;
+
+        const rootClasses = ClassNames(
+            UI_CLASS_NAME,
+            'label',
+            classes.root,
+            className,
+            {
+                'label-clickable': onClick,
+                'label-color-alert': color === 'alert',
+                'label-color-highlight': color === 'highlight',
+                'label-color-primary': color === 'primary',
+                'label-color-success': color === 'success',
+                'label-color-transparent': color === 'transparent',
+                'label-color-warning': color === 'warning',
+                'label-fluid': fluid,
+                'label-inverse': inverse,
+                'label-clearable': onClearClick,
+            },
+        );
+
+        return (
+            <div
+                className={rootClasses}
+                onClick={this.onClick}
+                onKeyDown={this.onKeyDown}
+                role="button"
+                style={style}
+                tabIndex={tabIndex}
+            >
+                <span className="label-inner-container">
+                    {children}
+
+                    {onClearClick && (
+                        <span
+                            className="label-clearable-button"
+                            onClick={this.onClearClick}
+                            onKeyDown={this.onClearKeyDown}
+                            role="button"
+                            tabIndex={-1}
+                        >
+                            <Icon
+                                compact
+                                inverse
+                                size="xxsmall"
+                                type="times"
```

Comment:
Should we put `title={false}` on this guy?

Reply:
Fixed.

Reply:
Looks good.  Resolving.